pkg/zinit/commands.go

```diff
@@ -304,3 +323,29 @@ func (c *Client) Forget(service string) error {
 func (c *Client) Kill(service string, sig Signal) error {
 	return c.cmd(fmt.Sprintf("kill %s %s", service, string(sig)), nil)
 }
+
+// Destroy a service completely (stop, forget and remove) if all filters are matched
+func (c *Client) Destroy(service string, filters ...Filter) error {
```

Comment:
I suggested also that the searching `the matching` and `the destroying` are to be done sepeartely. So you have a Match method, and the Destroy method that takes list of service names.

Reply:
Why you still provide the service name? this should be in the filters.